### ISSUE TITLE
Cannot use a sc.variable in a patch file to change the location of generated classes

### ISSUE THREAD

**Issue by thesoftwarejedi** [kamsar/Synthesis#27]
I'm trying to use a variable defined in a dev patch file that allows the dev to specify the source root folder.  We don't use the webroot as the source root.  Unicorn let's me do this.  With Synthesis it is passed along as a string.

```
<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
  <sitecore>
    <synthesis>
      <providers>
        <generatorParametersProvider type="Synthesis.Configuration.ConfigurationGeneratorParametersProvider, Synthesis">
          <ItemOutputPath>$(sourceRoot)/Website/Code/Synthesis/Synthesis.Model.Concrete.cs</ItemOutputPath>
          <InterfaceOutputPath>$(sourceRoot)/Website/Code/Synthesis/Synthesis.Model.cs</InterfaceOutputPath>
        </generatorParametersProvider>
      </providers>
    </synthesis>
  </sitecore>
</configuration> 
```

**Comment by kamsar** [comment]
This is caused because Sitecore does not expand variables in element bodies, only attributes.

For example,

`<InterfaceOutputPath foo="$(sourceRoot)">hello</InterfaceOutputPath>`

would resolve `sourceRoot`, whereas as you found

`<InterfaceOutputPath>$(sourceRoot)\hello</InterfaceOutputPath>`

will not expand. You can see expansion in the showconfig.aspx page - the variable is applied at the config provider level.

Unicorn gets it because you're injecting it on an attribute as opposed to a body.

Unfortunately without altering the configuration schema significantly and breaking backwards compatibility there is little I can do about this. It's Sitecore's rules that are being followed.

Now that said, since multiple configs are supported it would be fairly easy to alter the way the default config is registered and/or register your own config - which can expand whatever variables you like since at that point the registration is in C#.

**Comment by kamsar** [comment]
I decided to make a workaround anyway because you're probably not the last person to run into this issue.

See https://github.com/kamsar/Synthesis/commit/d96e1143f2219f0a983dcec14c7c807e81ed8f50

Essentially you can use variables in a setting that is prepended to all configs' output paths.

This will be in the next release.

**Comment by kamsar** [comment]
Fixed in 8.2.1
